apps/wallet/src/ui/app/pages/swap/utils.ts

```diff
@@ -1,16 +1,49 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { coinsMap } from '_app/hooks/useDeepBook';
+import { filterAndSortTokenBalances } from '_helpers';
+import { mainnetDeepBook, useActiveAddress, useCoinsReFetchingConfig } from '_hooks';
+import { useCoinMetadata, useFormatCoin } from '@mysten/core';
+import { useSuiClientQuery } from '@mysten/dapp-kit';
 
-export const DEFAULT_MAX_SLIPPAGE_PERCENTAGE = 0.5;
+export const DEFAULT_MAX_SLIPPAGE_PERCENTAGE = '0.5';
 export const FEES_PERCENTAGE = 0.03;
 
 export const initialValues = {
 	amount: '',
 	isPayAll: false,
-	quoteAssetType: coinsMap.USDC,
+	quoteAssetType: mainnetDeepBook.coinsMap.USDC,
 	allowedMaxSlippagePercentage: DEFAULT_MAX_SLIPPAGE_PERCENTAGE,
 };
 
 export type FormValues = typeof initialValues;
+
+export function useCoinTypeData(coinType: string | null) {
```

Comment:
I'm very unconvinced that this needs to be a custom hook, especially if you use `getBalance` instead of `getAllBalacnce`. The only real logic seems to be the filtering.